bin/update-ips.bash

```diff
@@ -224,9 +221,9 @@ get_swift_url() {
     log_error "Could not find Swift credentials in ${swift_config_option}"
     exit 1
   fi
-
-  os_token=$(grep -oP "x-subject-token:\s+\K\S+" "${headers}")
-  swift_url=$(echo "${response}" | jq -r '.token.catalog[] | select( .type == "object-store" and .name == "swift") | .endpoints[] | select(.interface == "public" and .region == "'"${swift_region}"'") | .url')
+  body=$(echo "${response}" | tr '\r\n' '!' | sed 's/!!/\n/g' | tail -n 1 | tr '!!' '\n')
+  os_token=$(echo "${response}" | grep -oP "x-subject-token:\s+\K\S+")
+  swift_url=$(echo "${body}" | jq -r '.token.catalog[] | select( .type == "object-store" and .name == "swift") | .endpoints[] | select(.interface == "public" and .region == "'"${swift_region}"'") | .url')
```

Comment:
This trick makes me a bit worried. Could there be a problem with existing exclamation marks in the response?

What about something like this?

```suggestion
{
  while read -d $'\r\n' header value; do
    [[ -z "${header}" ]] && break
    [[ "${header}" == "x-subject-token:" ]] && os_token="${value}"
  done

  swift_url=$(jq -r '.token.catalog[] | select( .type == "object-store" and .name == "swift") | .endpoints[] | select(.interface == "public" and .region == "'"${swift_region}"'") | .url')
} <<< "${response}"
```

This first reads all the headers, tries to find the token header and then passes the rest of the response (the body) to `jq`.

Ideally you would feed the output of `curl` directly to it rather than storing the response in a variable but don't know how much work that would be.

Note that I've not tested this so please try it out before committing it if you want to go with this solution instead. :)

Reply:
Yeah, it is always abit iffy replacing characters. But I will try your solution and see if it works :+1: